app/src/main/java/com/example/splendormobilegame/activities/GameActivity/GameActivity.java

```diff
@@ -312,13 +316,59 @@ public void onClick(DialogInterface dialog, int which) {
                 })
                 .show();
     }
-    public void setupCardsFirstTierRecyclerView(){
+
+    public void showDeckReserveDialog(UUID cardUuid) {
+        new MaterialAlertDialogBuilder(this)
+                .setTitle(getResources().getString(R.string.card_title))
+                .setMessage(getResources().getString(R.string.card_message))
+                .setNegativeButton(getResources().getString(R.string.cancel), new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        //hide dialog
+                    }
+                })
+                .setPositiveButton(getResources().getString(R.string.reserve_card), new DialogInterface.OnClickListener() {
+                    @Override
+                    public void onClick(DialogInterface dialog, int which) {
+                        // Respond to positive button press (right button)
+                        deckReservingController.reserveCard(cardUuid);
+                    }
+                })
+                .show();
+    }
+
+    private void setupReservingFromDeckButtons() {
+        binding.CardPile1CardView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                //reserve 1 tier card
+                showDeckReserveDialog(1);
```

Comment:
You have corrected this to provide number as a card tier, but `showDeckReserveDialog` still expects UUID as parameter.